sdks/java/extensions/sketching/src/main/java/org/apache/beam/sdk/extensions/sketching/ApproximateDistinct.java

```diff
@@ -0,0 +1,600 @@
+ * {@link ApproximateDistinctFn#withPrecision(int)} and
+ * {@link ApproximateDistinctFn#withSparseRepresentation(int)} methods.
+ *
+ * <pre>{@code
+ * PCollection<MyObject> input = ...;
+ * PCollection<HyperLogLogPlus> output = input.apply(Combine.globally(ApproximateDistinctFn
+ *                .<MyObject>create(new MyObjectCoder())
+ *                .withPrecision(18)
+ *                .withSparseRepresentation(24)));
+ * }</pre>
+ *
+ * <h4>Example 3 : use the {@link RetrieveCardinality} utility class</h4>
+ *
+ * <p>One may want to retrieve the cardinality as a long after making some advanced processing
+ * using the {@link HyperLogLogPlus} structure.
+ * <br>The {@link RetrieveCardinality} utility class provides an easy way to do so :
+ *
+ * <pre>{@code
+ * PCollection<MyObject> input = ...;
+ * PCollection<HyperLogLogPlus> hll = input.apply(Combine.globally(ApproximateDistinctFn
+ *                  .<MyObject>Create(new MyObjectCoder())
+ *                  .withSparseRepresentation(20)));
+ *
+ *  // Some advanced processing
+ *  PCollection<SomeObject> advancedResult = hll.apply(...);
+ *
+ *  PCollection<Long> cardinality = hll.apply(ApproximateDistinct.RetrieveCardinality.globally());
+ *  }</pre>
+ *
+ * <h4>Example 4 : use the {@link CombineFn} in a stateful {@link ParDo}</h4>
+ *
+ * <p>One may want to use the {@link ApproximateDistinctFn} in a stateful ParDo in order to
+ * make some processing depending on the current cardinality of the stream.
+ * <br>For more information about stateful processing see :
+ * <a>https://beam.apache.org/blog/2017/02/13/stateful-processing.html</a>
+ *
+ * <p>Here is an example of {@link DoFn} using an {@link ApproximateDistinctFn} as a
+ * {@link org.apache.beam.sdk.state.CombiningState} :
+ *
+ * <pre>{@code
+ * class StatefulCardinality<V> extends DoFn<V>, OutputT>> {
+ *   {@literal @}StateId("hyperloglog")
+ *   private final StateSpec<CombiningState<V, HyperLogLogPlus, HyperLogLogPlus>> indexSpec;
+ *
+ *   public StatefulCardinality(ApproximateDistinctFn<V> fn) {
+ *     indexSpec = StateSpecs.combining(fn);
+ *   }
+ *
+ *   {@literal @}ProcessElement
+ *   public void processElement(
+ *      ProcessContext context,
+ *      {@literal @}StateId("hllSketch")
+ *      CombiningState<V, HyperLogLogPlus, HyperLogLogPlus> hllSketch) {
+ *     long current = MoreObjects.firstNonNull(hllSketch.getAccum().cardinality(), 0L);
+ *     hllSketch.add(context.element());
+ *     context.output(...);
+ *     }
+ * }
+ * }</pre>
+ *
+ * <p>Then the {@link DoFn} can be called like this :
+ * <pre>{@code
+ * PCollection<Object> input = ...;
+ * ApproximateDistinctFn myFn = ApproximateDistinctFn.create(new ObjectCoder());
+ * PCollection<OutputT> = input.apply(ParDo.of(new StatefulCardinality(myFn)));
+ * }</pre>
+ *
+ */
+public final class ApproximateDistinct {
+
+  /**
+   * The {@link PTransform} takes an input {@link PCollection} of objects and returns a
```

Comment:
"can be queried" is now moot, the documentation can be made a lot more concise, e.g.: "Computes the approximate number of distinct elements in the input `PCollection<InputT>` as a `PCollection<Long>`".